server/src/main/java/org/elasticsearch/ExceptionsHelper.java

```diff
@@ -176,9 +176,17 @@ public static String formatStackTrace(final StackTraceElement[] stackTrace) {
     }
 
     public static IOException unwrapCorruption(Throwable t) {
-        return (IOException) unwrap(t, CorruptIndexException.class,
-                                       IndexFormatTooOldException.class,
-                                       IndexFormatTooNewException.class);
+        IOException corruptionException =
+            (IOException) unwrap(t, CorruptIndexException.class, IndexFormatTooOldException.class, IndexFormatTooNewException.class);
+        if (corruptionException == null && t != null) {
+            for (Throwable suppressed : t.getSuppressed()) {
```

Comment:
do we need to check suppressed exceptions of `Throwable.getCause` etc?

Reply:
Right! I guess we should will add that together with tests in a sec :)